pkg/util/names/names.go

```diff
@@ -1,9 +1,23 @@
 package names
 
 import (
+	"crypto/sha1"
+	"encoding/hex"
 	"fmt"
 )
 
 func GenJobName(jsName, rjobName string, jobIndex int) string {
 	return fmt.Sprintf("%s-%s-%d", jsName, rjobName, jobIndex)
 }
+
+// JobHashKey returns the SHA1 hash of the namespaced job name (i.e. <namespace>/<jobName>).
+func JobHashKey(ns string, jobName string) string {
```

Comment:
why did you need to move this here?

Reply:
To avoid an import cycle between v1alpha2 package and controllers package

Reply:
but you have the key as a label on the pod, so only the jobset controller should need this function, in all other places we just need to look it up from the label